packages/flutter_tools/lib/src/base/dds.dart

```diff
@@ -4,112 +4,329 @@
 
 import 'dart:async';
 
-import 'package:dds/dds.dart' as dds;
 import 'package:meta/meta.dart';
 
-import 'common.dart';
-import 'context.dart';
+import '../convert.dart';
+import '../device.dart';
 import 'io.dart' as io;
 import 'logger.dart';
+import 'platform.dart' as p;
 
-// TODO(fujino): This should be direct injected, rather than mutable global state.
-@visibleForTesting
-Future<dds.DartDevelopmentService> Function(
+/// A representation of the current DDS state including:
+///
+/// - The process the external DDS instance is running in
+/// - The service URI DDS is being served on
+/// - The URI DevTools is being served on, if applicable
+/// - The URI DTD is being served on, if applicable
+typedef DartDevelopmentServiceInstance = ({
+  io.Process? process,
+  Uri? serviceUri,
+  Uri? devToolsUri,
+  Uri? dtdUri,
+});
+
+/// The default DDSLauncherCallback used to spawn DDS.
+Future<DartDevelopmentServiceInstance> defaultStartDartDevelopmentService(
   Uri remoteVmServiceUri, {
-  bool enableAuthCodes,
-  bool ipv6,
+  required bool enableAuthCodes,
+  required bool ipv6,
+  required bool enableDevTools,
+  required List<String> cachedUserTags,
   Uri? serviceUri,
-  List<String> cachedUserTags,
-  dds.UriConverter? uriConverter,
-}) ddsLauncherCallback = dds.DartDevelopmentService.startDartDevelopmentService;
+  String? google3WorkspaceRoot,
+  Uri? devToolsServerAddress,
+}) async {
+  final String exe = const p.LocalPlatform().executable;
+  final io.Process process = await io.Process.start(
+    exe,
```

Comment:
Can you instead require the path to the dart executable as a function argument?

Then at the callsite of this function, you can get the path via something like:

```dart
final String dartPath = globals.artifacts.getArtifactPath(Artifact.engineDartBinary);
```

Reply:
Done.